megamek/src/megamek/client/ui/swing/CommonMenuBar.java

```diff
@@ -217,606 +134,294 @@ public CommonMenuBar(Client parent) {
-        menu.add(viewZoomOut);
+        viewResetWindowPositions = createMenuItem(menu, getString("viewResetWindowPos"), VIEW_RESET_WINDOW_POSITIONS);
+        //TODO: show minimap should be a checkbox
+        viewMiniMap = createMenuItem(menu, getString("viewMiniMap"), VIEW_MINI_MAP, KeyEvent.VK_M);
+        //TODO: show unit overview should be a checkbox
+        viewUnitOverview = createMenuItem(menu, getString("viewUnitOverview"), VIEW_UNIT_OVERVIEW, KeyEvent.VK_U);
+        viewZoomIn = createMenuItem(menu, getString("viewZoomIn"), VIEW_ZOOM_IN);
+        viewZoomOut = createMenuItem(menu, getString("viewZoomOut"), VIEW_ZOOM_OUT);
         menu.addSeparator();
-        toggleIsometric = new JCheckBoxMenuItem(Messages
-                .getString("CommonMenuBar.viewToggleIsometric")); //$NON-NLS-1$
-        toggleIsometric.addActionListener(this);
-        toggleIsometric.setState(GUIPreferences.getInstance().getBoolean("UseIsometric")); //$NON-NLS-1$
-        toggleIsometric.setActionCommand(ClientGUI.VIEW_TOGGLE_ISOMETRIC);
-        menu.add(toggleIsometric);
-        toggleFovDarken = new JCheckBoxMenuItem(Messages
-                .getString("CommonMenuBar.viewToggleFovDarken")); //$NON-NLS-1$
-        toggleFovDarken.addActionListener(this);
-        toggleFovDarken.setState(GUIPreferences.getInstance().getBoolean("FovDarken")); //$NON-NLS-1$
-        toggleFovDarken.setActionCommand(ClientGUI.VIEW_TOGGLE_FOV_DARKEN);
-        toggleFovDarken.setToolTipText(Messages.getString("CommonMenuBar.viewToggleFovDarkenTooltip"));
-        menu.add(toggleFovDarken);
-        toggleFovHighlight = new JCheckBoxMenuItem(Messages
-                .getString("CommonMenuBar.viewToggleFovHighlight")); //$NON-NLS-1$
-        toggleFovHighlight.setState(GUIPreferences.getInstance().getBoolean("FovHighlight")); //$NON-NLS-1$
-        toggleFovHighlight.addActionListener(this);
-        toggleFovHighlight.setActionCommand(ClientGUI.VIEW_TOGGLE_FOV_HIGHLIGHT);
-        menu.add(toggleFovHighlight);
-        toggleFieldOfFire = new JCheckBoxMenuItem(Messages
-                .getString("CommonMenuBar.viewToggleFieldOfFire")); //$NON-NLS-1$
-        toggleFieldOfFire.addActionListener(this);
+        toggleIsometric = createCbxMenuItem(menu, getString("viewToggleIsometric"), VIEW_TOGGLE_ISOMETRIC);
+        toggleIsometric.setState(GUIPreferences.getInstance().getBoolean(GUIPreferences.USE_ISOMETRIC)); 
+        toggleFovDarken = createCbxMenuItem(menu, getString("viewToggleFovDarken"), VIEW_TOGGLE_FOV_DARKEN);
+        toggleFovDarken.setState(GUIPreferences.getInstance().getBoolean(GUIPreferences.FOV_DARKEN)); 
+        toggleFovDarken.setToolTipText(getString("viewToggleFovDarkenTooltip"));
+        toggleFovHighlight = createCbxMenuItem(menu, getString("viewToggleFovHighlight"), VIEW_TOGGLE_FOV_HIGHLIGHT);
+        toggleFovHighlight.setState(GUIPreferences.getInstance().getBoolean("FovHighlight"));
```

Comment:
Can we use the proper constant here, instead of getting the boolean based on an inline string?